server/vertx-server/src/main/scala/sttp/tapir/server/vertx/VertxFutureServerInterpreter.scala

```diff
@@ -26,7 +26,9 @@ trait VertxFutureServerInterpreter extends CommonServerInterpreter with VertxErr
     *   A function, that given a router, will attach this endpoint to it
     */
   def route[A, U, I, E, O](e: ServerEndpoint[VertxStreams with WebSockets, Future]): Router => Route = { router =>
-    mountWithDefaultHandlers(e)(router, extractRouteDefinition(e.endpoint), vertxFutureServerOptions)
+    val routeDef = extractRouteDefinition(e.endpoint)
+    optionsRoute(e)(router, routeDef).foreach(_.handler(endpointHandler(e)))
```

Comment:
I think this would need some more explanation, as to why it's needed here and what's the purpose. Won't we run the routes twice?

Reply:
Having only i.e. `GET /path` is not sufficient 🫨 , so by adding its counterpart `OPTIONS /path` I am explaining to Vert.x : please accept my preflight request instead of rejecting it quickly